src/orientationUtils/RelativeDataHandler.java

```diff
@@ -0,0 +1,27 @@
+package orientationUtils;
+
+import java.util.function.Supplier;
+
+public class RelativeDataHandler {
+
+	private Supplier<Double> relativeDataSupplier;
+
+	private double lastValue;
+	private double currentValue;
+
+	public RelativeDataHandler(Supplier<Double> relativeValueSupplier) {
+		this.relativeDataSupplier = relativeValueSupplier;
+		lastValue = 0;
+		currentValue = 0;
+	}
+
+	public double getLastDifference() {
+		double result = currentValue - lastValue;
+
+		lastValue = currentValue;
+		currentValue = relativeDataSupplier.get();
+
+		return result;
+	}

```

Comment:
this will return the difference between the k-1 and k-2 measurements, and not the difference between k and k-1

Reply:
Nice catch.